chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb

```diff
@@ -25,6 +25,13 @@ if [[ -f /usr/bin/neutron-ha-tool ]] ; then
     OS_AUTH_URL="<%= @os_auth_url_v2 %>" neutron-ha-tool --l3-agent-evacuate $(hostname)
 fi
 
+# Even if agents are migrated, we want to explicitly destroy the router namespaces
+# because in some upgrade cases, the routers remain active when there are no
+# valid destinations for the migration
```

Comment:
@rhafer btw, I've moved this code to https://github.com/crowbar/crowbar-core/pull/779